src/shared/components/common/state-radios.tsx

```diff
@@ -0,0 +1,62 @@
+import { I18NextService } from "@services/index";
+import { randomStr } from "@utils/helpers";
+import classNames from "classnames";
+import { NoOptionI18nKeys } from "i18next";
+import { linkEvent } from "inferno";
+import { Component } from "inferno";
+
+export interface State {
+  value: string;
+  i18n: NoOptionI18nKeys;
+}
+
+export interface StateRadioProps {
+  className?: string;
+  allStates: State[];
+  currentState: string;
+  onClickHandler(val: string): void;
+}
+
+export class StateRadio extends Component<StateRadioProps, object> {
```

Comment:
Its fine for this to be a component, but in the future, if it only has props and no state, then you can just use a functional component.

https://www.infernojs.org/docs/guides/components

Check out `src/shared/components/post/post-action-bar.tsx` for an example.